monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/NotificationManager.java

```diff
@@ -66,12 +77,20 @@ public void deleteRecipient(String email) {
 		return Arrays.asList(dbConnection.getAllRecipients());
 	}
 
-	@Override
-	public void sendNotificationEmails(IApp app, AppStatus previousStatus) {
-		INotificationRecipient[] recipients = dbConnection.getAllRecipients();
-		for (INotificationRecipient recipient : Arrays.asList(recipients)) {
-			emailManager.sendAppStatusNotificationEmail(recipient.getEmail(), app, previousStatus.toString(),
-					app.getLastAppTest().getStatus().toString());
-		}
-	}
+	@Override //change here
+    public void sendNotificationEmails(IApp app, AppStatus previousStatus) {
+	    List<INotificationRecipient> recipients;
+        if (app.getRecipients().isEmpty()) {
+            recipients = Arrays.asList(dbConnection.getAllRecipients());
+        }
+        else {
+            recipients = new ArrayList<>();
+            for (INotificationRecipient recipient : app.getRecipients())
+                recipients.add(recipient);
```

Comment:
- please follow our style guide
- also, why do you add all recipients  into a new list?